### PR TITLE
Support `from typing import ClassVar`

### LINKED ISSUE #361: Support ClassVar string annotations
The following doesn't work in Python 3.7

```
from __future__ import annotations
import attr
from typing import ClassVar

@attr.dataclass
class A:
    x: ClassVar[int]

a = A()
```

Because `ClassVar[int]` will be `'ClassVar[int]` in `A.__annotations__` and so `return str(annot).startswith("typing.ClassVar")` returns `False`.

You can repro without 3.7 using this:

```
import attr
from typing import ClassVar

@attr.dataclass
class A:
    x: 'ClassVar[int]'

a = A()
```

Note: It's weird I'm not giving it a value but if I do then attrs will think it's a default value and the code won't error out.

Oh and the following works fine:
```
import attr
import typing

@attr.dataclass
class A:
    x: 'typing.ClassVar[int]'

a = A()
```

### PR DESCRIPTION
**Pull request by ambv**
Also updated the docstring to reflect why exactly we're doing what we're doing.
The previous comment was incorrect (`typing` is already imported in
applications using annotations).

I added `t.ClassVar` as well which is the only third alternative import I found
in use for typing-related classes.

Fixes #361

### CODE REVIEW

File: src/attr/_make.py [hunk 1]
```diff
@@ -229,13 +229,13 @@ class MyClassAttributes(tuple):
 
 
 def _is_class_var(annot):
-    """
-    Check whether *annot* is a typing.ClassVar.
+    """Check whether *annot* is a typing.ClassVar.
```
Comment (hynek) [inline src/attr/_make.py]: This is not how we write docstrings around here! 👨‍🏫

File: src/attr/_make.py [hunk 2]
```diff
@@ -229,13 +229,13 @@ class MyClassAttributes(tuple):
...
 
-    The implementation is gross but importing `typing` is slow and there are
-    discussions to remove it from the stdlib alltogether.
+    The string comparison hack is used to avoid evaluating all string
+    annotations which would put attrs-based classes at a performance
+    disadvantage compared to plain old classes.
     """
-    return str(annot).startswith("typing.ClassVar")
+    return str(annot).startswith(("typing.ClassVar", "t.ClassVar", "ClassVar"))
```
Comment (hynek) [inline src/attr/_make.py]: I think we should amend our tests to cover the new cases?

Comment (ambv) [reply src/attr/_make.py]: Yes.

**Comment by hynek** [comment]
🏓

**Comment by ambv** [comment]
(updated)

**Comment by hynek** [comment]
Thanks everyone! At least we can feel bad about that hack *together*. ;)

**Pull request merged**